static-instrumenter/agent-instrumenter/src/main/java/io/opentelemetry/contrib/staticinstrumenter/util/AgentExtractor.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.staticinstrumenter.util;
+
+import io.opentelemetry.contrib.staticinstrumenter.util.path.AgentPathGetter;
+import io.opentelemetry.contrib.staticinstrumenter.util.path.SimplePathGetter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Enumeration;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import org.apache.commons.io.IOUtils;
+
+/**
+ * Extracts the agent to a temporary directory. This is done to enable simple loading of agent
+ * classes that need to be modified.
+ */
+public final class AgentExtractor {
+
+  private final Path extractedAgent;
+
+  public AgentExtractor(TmpDirManager tmpDirManager) throws IOException {
+    this.extractedAgent = tmpDirManager.createDir("opentelemetry-javaagent");
+  }
+
+  /**
+   * Extracts the agent JAR entry to a temporary directory.
+   *
+   * @param agentFile File object representing the OpenTelemetry javaagent file
+   * @throws IOException If process of writing to file encounters problems
```

Comment:
Nit: instead of describing the method param/thrown exception (which are kinda obvious), WDYT about mentioning that the `inst/` & `.classdata` convention is dropped, so that agent classes can be "normally" loaded?